frontend/src/pages/projects/screens/detail/pipelines/PipelinesList.tsx

```diff
@@ -29,6 +33,10 @@ const PipelinesList: React.FC = () => {
   }
 
   if (pipelines.length === 0) {
+    // Call setIsPipelinesEmpty only if it is provided
+    if (setIsPipelinesEmpty) {
+      setIsPipelinesEmpty(true);
```

Comment:
For future reference, callbacks should never happen during the render cycle of a component but instead moved to a `useEffect`.